### ISSUE TITLE
social media share previews do not pull custom favicon for preview

### ISSUE THREAD

**Issue by jcklpe** [nextcloud/server#18435]
### Steps to reproduce
1. copy paste a public link share to social media such as slack or facebook
2. social media post editor will autogenerate a preview image using favicon-fb.png
3. favicon-fb.png isn't the custom favicon associated with a themed NextCloud instance

### Expected behaviour
The logo set in theming/ the favicon which is generated by the process, should appear as a preview image instead of the default nextcloud logo. 

### Actual behaviour
It shows the default nextcloud logo:

![image](https://user-images.githubusercontent.com/9451726/70950995-ce966e80-2027-11ea-8016-fde6b2dc676e.png)

### Server configuration

**Operating system:** Ubuntu 18.04

**Web server:** nginx

**Database:** postgresSQL

**PHP version:** 7.3

**Nextcloud version:** 17

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** I installed using this ansible script which I modified slightly: https://github.com/ReinerNippes/nextcloud

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```

ONLYOFFICE
3.0.2
Official
Accessibility
1.3.0
Official
Activity
2.10.1
Official
Auditing / Logging
1.7.0
Official
Brute-force settings
1.4.0
Official
Collaborative tags
1.7.0
Official
Comments
1.7.0
Official
Custom CSS
1.4.0
Deleted files
1.7.0
Official
External storage support
1.8.0
Official
Federation
1.7.0
Official
File access control
1.7.0
Official
File sharing
1.9.0
Official
Gallery
18.4.0
Official
Group folders
5.0.4
Official
Log Reader
2.2.0
Official
Mind Map
0.0.15
Monitoring
1.7.0
Official
Nextcloud announcements
1.6.0
Official
Notifications
2.5.0
Official
Password policy
1.7.0
Official
PDF viewer
1.6.0
Official
Privacy
1.1.0
Official
Recommendations
0.5.0
Official
Right click
0.15.1
Official
Share by mail
1.7.0
Official
ShareRenamer
2.7.3
Support
1.0.1
Official
Text
1.1.1
Official
Theming
1.8.0
Official
Update notification
1.7.0
Official
Versions
1.10.0
Official
Video player
1.6.0
Official
Viewer
1.2.0
Official 
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** firefox

**Operating system:** Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```

Error | PHP | rename(//mnt/work/jackalope/resume/public-resumes/debug.log,//var/nc-data/aslan/files_trashbin/files/debug.log.d1576432183):  Operation not permitted at  /var/www/nextcloud/lib/private/Files/Storage/Local.php#289 |  
-- | -- | -- | --



```
</details>

I think it's just that the favicon-fb.png isn't getting updated. Seems like it's probably a pretty simple issue to fix. I tried searching the favicon-fb.png in the source code and found only one place where it was mentioned. All other instance of the favicon work fine though.

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/blob/5bf3d1bb384da56adbf205752be8f840aac3b0c5/apps/files_sharing/lib/Controller/ShareController.php#L415-L416

We need to fetch the favicon from theming app here.

**Comment by kesselb** [comment]
```
Index: apps/files_sharing/lib/Controller/ShareController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/files_sharing/lib/Controller/ShareController.php	(revision 0ddb9c01362a74662a4f062683c82c1594c7da01)
+++ apps/files_sharing/lib/Controller/ShareController.php	(date 1576941085287)
@@ -412,7 +412,7 @@
 				}
 			}
 		} else {
-			$shareTmpl['previewImage'] = $this->urlGenerator->getAbsoluteURL($this->urlGenerator->imagePath('core', 'favicon-fb.png'));
+			$shareTmpl['previewImage'] = $this->urlGenerator->getAbsoluteURL($this->urlGenerator->imagePath('', 'favicon.ico'));
 			$ogPreview = $shareTmpl['previewImage'];
 		}
```

Does it work if you alter the code like above? But the favicon is probably to small with 32x32.

**Comment by jcklpe** [comment]
@kesselb I'm a little hesistant to dig directly into the php code for my nextcloud server, as in my experience, doing so triggers security issues right? It causes the NC server to fail integrity checks etc right?

**Comment by kesselb** [comment]
Integrity check will complain yes.

**Comment by jcklpe** [comment]
@kesselb  Is there a way for me to help push this issue forward without doing that? My NC server is my personal server, and last time I did something that caused an integrity check complaint it plagued me for like 10 months and literally nothing I could do would get it to stop, and so I'm very hesitant about the idea of monkeying around with the internals of NC. 

It's not a "production" environment per se because it's just my personal server that I run out of my closet but I also don't have a staging or dev environment to test this issue. I'm doing this entirely as an individual. 

This issue shouldn't be too hard for pretty much anyone to replicate though. Literally one just needs to change the default theming for NC by uploading a new logo and changing the primary color scheme and then share a link to a slack channel or etc. I just installed this version of the server as a completely fresh install (which I did because in part the aforementioned issues with the file integrity check constantly giving me problems).
